test/utils_test.rb

```diff
@@ -0,0 +1,171 @@
+require File.expand_path(File.join(File.dirname(__FILE__), "test_helper"))
+
+class RubySamlTest < Minitest::Test
+
+  describe "#format_cert" do
+    it "return empty string when the cert is an empty string" do
+      cert = ""
+      assert_equal cert, OneLogin::RubySaml::Utils.format_cert(cert, true)
+      assert_equal cert, OneLogin::RubySaml::Utils.format_cert(cert, false)
+    end
+
+    it "return formatted cert with head from cert with head" do
+      cert = ruby_saml_cert_text
+      formatted_cert = OneLogin::RubySaml::Utils.format_cert(cert, true)
+      assert cert.include? "-----BEGIN CERTIFICATE-----"
+      assert cert.include? "-----END CERTIFICATE-----"
+      assert_equal cert, formatted_cert
+      assert formatted_cert.include? "-----BEGIN CERTIFICATE-----"
+      assert formatted_cert.include? "-----END CERTIFICATE-----"
+    end
+
+    it "return formatted cert with head from cert without head" do
```

Comment:
@Lordnibbler I can see at OpenSSL any method to format the cert/private_key.

Imagine those scenarios:

A user copy a cert from a website where the public cert file is printed, but there are extra new lines, when he paste the cert in the ruby-saml setting file, the 
```OpenSSL::X509::Certificate.new(cert_string)```
command will fail. If we previously call the format_cert method, the user will not experience any problem.

I added here a test so you can review what I said:
https://github.com/onelogin/ruby-saml/commit/da8c9ec1679cbe66ec6be5411b48a9faa63f9a28